### ISSUE TITLE
FEATURE: Add support for lsp related file operations

### ISSUE THREAD

**Issue by ghostbuster91** [nvim-neo-tree/neo-tree.nvim#1050]
### Did you check the docs?

- [X] I have read all the docs.

### Is your feature request related to a problem? Please describe.

Example feature:
- [workspace/WillRename](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_willRenameFiles)

### Describe the solution you'd like.

Integrate with nvim-lsp-file-operations, which is a Neovim plugin that adds support for file operations using built-in [LSP support](https://neovim.io/doc/user/lsp.html).

**Comment by nhat-vo** [comment]
If I'm not wrong then this is similar to #593, is that correct?

**Comment by ghostbuster91** [comment]
Not really, the one that you mentioned is about displaying workspace symbols on a tree while this one is about performing some lsp-based refactorings when filesystem tree is modified. I forgot to link but this has been already implemented for nvim-tree here https://github.com/antosha417/nvim-lsp-file-operations and the readme there does probably a better job explaining that feature.

**Comment by cseickel** [comment]
It is kind of a duplicate of #47. I've been using the solution posted by @gelio in [this comment](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/308#issuecomment-1304765940) and it has worked great for typescript files, but it would definitely be better to natively integrate with **nvim-lsp-file-operations** as an official solution.

**Comment by cseickel** [comment]
I just submitted a PR to add Neo-tree support to **nvim-lsp-file-operations**:
https://github.com/antosha417/nvim-lsp-file-operations/pull/14

**Comment by cseickel** [comment]
That PR was merged, but I will leave this issue open as a reminder to mention it in our README.
